rotkehlchen/chain/manager.py

```diff
@@ -430,13 +430,13 @@ def set_dot_rpc_endpoint(self, endpoint: str) -> Tuple[bool, str]:
     def activate_premium_status(self, premium: Premium) -> None:
         self.premium = premium
         for _, module in self.iterate_modules():
-            if getattr(module, 'premium', None):
+            if getattr(module, 'premium', None) is None:
```

Comment:
This is not gonna work in all cases.

What happens if you want to activate `premium` and the `module`'s premium has a previous instantiation of a `Premium` object for some reason?

So this scenario:

- `premium` argument is the newly initialized `Premium`
- `module.premium` is not `None`, but an old initialized `Premium` object.

In that case what you do is you don't change it. But you should.

Reply:
When this is a possible case I believe it shouldn't happen since we should always use the `deactivate_premium_status` method and there shouldn't be two instances of the premium object around. Since I used conditionals there I'll change this to make sure that is always removed